File: design-pattern-go/strategy/hand.go
package strategy

type HandKind struct {
	ja string
	id int
}

var (
	HAND_KIND_ROCK     = HandKind{ja: "グー", id: 0}
	HAND_KIND_SCISSORS = HandKind{ja: "チョキ", id: 1}
	HAND_KIND_PAPER    = HandKind{ja: "パー", id: 2}
)

func (h *HandKind) GetJa() string {
	return h.ja
}

func (h *HandKind) GetId() int {
	return h.id
}

func GetHandKindById(id int) HandKind {
	if id == HAND_KIND_ROCK.id {
		return HAND_KIND_ROCK
	} else if id == HAND_KIND_SCISSORS.id {
		return HAND_KIND_SCISSORS
	} else {
		return HAND_KIND_PAPER
	}
}

type Hand struct {
	kind HandKind
}

func NewHand(kind HandKind) *Hand {
	return &Hand{
		kind,
	}
}

func (h *Hand) IsStrongerThan(opponent *Hand) bool {
	return h.fight(opponent) == 1
}

func (h *Hand) IsWeakerThan(opponent *Hand) bool {
	return h.fight(opponent) == -1
}

func (h *Hand) fight(opponent *Hand) int {
	if h.kind.id == opponent.kind.id {
		return 0
	} else if (h.kind.id+1)%3 == opponent.kind.id {
		return 1
	} else {
		return -1
	}
}

File: design-pattern-python/design_pattern_python/strategy/hand.py
from enum import Enum


class HandKind(Enum):
    ROCK = ("グー", 0)
    SCISSORS = ("チョキ", 1)
    PAPER = ("パー", 2)

    def __init__(self, ja: str, id: int) -> None:
        self.ja = ja
        self.id = id

    @classmethod
    def get_hand_kind_by_id(cls, id: int) -> "HandKind":
        if id == 0:
            return HandKind.ROCK
        elif id == 1:
            return HandKind.SCISSORS
        elif id == 2:
            return HandKind.PAPER
        else:
            raise Exception("input id is out of range!!")


class Hand:
    def __init__(self, kind: HandKind) -> None:
        self.kind = kind

    def is_stronger_than(self, opponent: "Hand") -> bool:
        return self.__fight(opponent) == 1

    def is_weaker_than(self, opponent: "Hand") -> bool:
        return self.__fight(opponent) == -1

    def __fight(self, opponent: "Hand") -> int:
        if self.kind == opponent.kind:
            return 0
        elif (self.kind.id + 1) % 3 == opponent.kind.id:
            return 1
        else:
            return -1
